transports/janus_rabbitmq.c

```diff
@@ -1024,13 +1035,19 @@ void *janus_rmq_out_thread(void *data) {
 		return NULL;
 	}
 	JANUS_LOG(LOG_VERB, "Joining RabbitMQ out thread\n");
+	guint rmq_reconnect_backoff = rmq_reconnect_backoff_initial;
 	while(!rmq_client->destroy && !g_atomic_int_get(&stopping)) {
 
 		while (!rmq_client->connected) {
```

Comment:
This should take into account whether the server is shutting down or not. The `while` outside does it, but the inner `while` doesn't, and so won't break if in between `g_usleep` calls the server has been closed. Maybe it's simpler to turn the inner `while` to an `if`, and do a `continue` after the sleep and the reconnection timer update, a bit like you do in `janus_rmq_in_thread`?

Reply:
Good point!